chainstate/src/detail/error.rs

```diff
@@ -49,6 +52,8 @@ pub enum BlockError {
     DatabaseCommitError(Id<Block>, usize, chainstate_storage::Error),
     #[error("Block proof calculation error for block: {0}")]
     BlockProofCalculationError(Id<Block>),
+    #[error("Error while flushing state to storage: {0}")]
+    FlushError(#[from] TransactionVerifierStorageError),
```

Comment:
I would call this TransactionVerifierError?

Reply:
why? `FlushError` sounds more precise about what went wrong

Reply:
Because flushing is an implementation detail.